### PR TITLE
[cxx-interop] Add basic C++ stdlib overlay

### PR DESCRIPTION
**Pull request by egorzhdan**
This will allow us to add Swift conformances to C++ standard library types, and improve usability of the C++ standard library from Swift.

### CODE REVIEW

File: test/Interop/Cxx/stdlib/overlay.swift [hunk 1]
```diff
@@ -0,0 +1,21 @@
+// RUN: %target-run-simple-swift(-I %S/Inputs -Xfrontend -enable-experimental-cxx-interop)
```
Comment (zoecarver) [inline test/Interop/Cxx/stdlib/overlay.swift:1]: Nit: maybe it makes more sense to have this be `Cxx/stdlib/overlay/string.swift` or `Cxx/overlay/string.swift`. WDYT?

**Review by zoecarver — APPROVED** [review]
Super exciting!

File: test/Interop/Cxx/stdlib/overlay.swift (hunk 1 above)
Comment (egorzhdan) [reply test/Interop/Cxx/stdlib/overlay.swift]: Yeap, let's make it `Cxx/stdlib/overlay/string.swift`

**Comment by egorzhdan** [comment]
@swift-ci please smoke test

**Comment by egorzhdan** [comment]
@swift-ci please smoke test

**Comment by egorzhdan** [comment]
@swift-ci please smoke test macOS

**Comment by egorzhdan** [comment]
@swift-ci please smoke test Linux

**Comment by egorzhdan** [comment]
@swift-ci please smoke test

**Comment by egorzhdan** [comment]
@swift-ci please smoke test

**Comment by egorzhdan** [comment]
@swift-ci please smoke test

**Comment by egorzhdan** [comment]
@swift-ci please smoke test

**Review by zoecarver — APPROVED** [review]

**Comment by egorzhdan** [comment]
@swift-ci please smoke test

**Comment by egorzhdan** [comment]
I added one more commit with a general C++ interop change, not specific to the overlay, to fix `verify_all_overlays.py`.
@zoecarver can you please take another look?

**Comment by egorzhdan** [comment]
Discussed offline last week, will merge this as soon as the tests pass (re-rerunning to make sure libswift doesn't break).

**Comment by egorzhdan** [comment]
@swift-ci please smoke test

**Comment by egorzhdan** [comment]
@swift-ci please test Windows

**Comment by egorzhdan** [comment]
@swift-ci please smoke test

**Comment by egorzhdan** [comment]
@swift-ci please smoke test

**Pull request merged**